src/charm.py

```diff
@@ -314,6 +312,14 @@ def peers(self) -> Optional[Relation]:
         """
         return self.model.get_relation(Config.Relations.PEERS)
 
+    @property
+    def peers_units(self) -> list[Unit]:
```

Comment:
BTW this was an excellent find and an excellent fix. I would be curious to see where else this sort of logic could be beneficial

Reply:
Thanks a lot!
I'll take some time to see if there are similar cases where we should do the same and if that's the case, I'll add all those functions :)